lib/api/apiUtils/object/setUpCopyLocator.js

```diff
@@ -63,10 +62,8 @@ function setUpCopyLocator(sourceObjMD, rangeHeader, log) {
                 };
         }
         if (range) {
-            dataLocator =
-                routesUtils.setPartRanges(dataLocator, range);
-            copyObjectSize = Math.min(sourceSize - range[0],
-                range[1] - range[0] + 1);
+            dataLocator = routesUtils.setPartRanges(dataLocator, range);
+            copyObjectSize = range[1] - range[0] + 1;
```

Comment:
It was, the returned range is guaranteed to not go beyond the object size, that is provided as a parameter.